src/main/java/org/eclipse/yasson/internal/model/PropertyModel.java

```diff
@@ -117,7 +121,7 @@ public PropertyModel(ClassModel classModel, Property property, JsonbContext json
             return new AdaptedObjectSerializer<>(classModel, customization.getSerializeAdapterBinding());
         }
         if (customization.getSerializerBinding() != null) {
-            return new UserSerializerSerializer<>(classModel, customization.getSerializerBinding().getJsonbSerializer());
+            return new UserSerializerSerializer<>(customization.getSerializerBinding().getJsonbSerializer(), new ObjectSerializer<>(null, propertyType, classModel));
```

Comment:
we can't assume that the ObjectSerializer will be an appropriate default serializer in all cases, for example a custom serializer for primitives or arrays might not work.
Can you add some test scenarios for a recursive serializer on array and primitive types? I am thinking it may be more appropriate to call `DefaultSerializers.getInstance().findValueSerializerProvider(Class)` to select the proper default serializer